File: examples/perf_comparison/alloc.py
class Point:
    def __init__(self, x, y, z):
        self.x = x
        self.y = y
        self.z = z

i = 0
while i < 1000000:
    p = Point(i, i + 1, i + 2)
    i += 1
print(i)

File: examples/perf_comparison/alloc.lua
local function Point(x, y, z)
    return {x = x, y = y, z = z}
end

local i = 0
while i < 1000000 do
    local p = Point(i, i + 1, i + 2)
    i = i + 1
end
print(i)
